Support UBL.be format

JorisDebonnet opened the pull request:
Hello! This pull request contains quite a few unrelated changes; I needed all of it to be able to (mostly) support UBL.be's format (while at the same time also supporting, as much  as possible from a few other formats). 

The only breaking change is that I've had to change the default unitCode for InvoiceLines in order to make it match the default unitCode for Price objects (... but it really makes no sense having different defaults for them).

Everything else should be backwards compatible. A few added attributes may be unneeded in some formats, but that should not result in any error; at most a warning.

If you believe this is fine, I will fill in the next-release.md. If not, no problem, then I'll just leave this in our own fork :)

BBrunekreeft commented on src/AdditionalDocumentReference.php:
```diff
@@ -47,6 +48,24 @@ public function setDocumentType(string $documentType): AdditionalDocumentReferen
+     */
+    public function getDocumentDescription(): ?string
+    {
+        return $this->documentType;
```
Shouldn't this be `return $this->documentDescription;`?

JorisDebonnet replied:
Absolutely, fixed it. I admit I did not add any automated tests or anything... I don't use any of the getters myself.

(once all done, I can clean up the commits a bit to make them more compact)

brtdv commented:
Hi guys, 

Thanks, I'll take a look ASAP and merge the changes into a new release 🙌

JorisDebonnet commented:
I just cleaned up the commits a little and added documentation for the changes.

brtdv commented:
Hi @JorisDebonnet,

Thank you for your PR. Sorry for the delay in merging this. 

I checked your changes and did some small formatting tweaks so that code formatting is consistent across the library.

I did however bump into two problems. When running the `composer test` to run all the currently set up unit tests, the `EN16931Test` fails with the following output:

```
*** [UBL-CR-660]-A UBL invoice should not include the Country Identification code listID
*** [UBL-CR-663]-A UBL invoice should not include the unitCodeListID
```

Therefore, to prevent other users' EN16931 UBL's from having unexpected errors, I cannot merge these changes right now, and we need to make sure these checks pass. 

1. Could it be the added listID attribute is not necessary for (Belgian) UBL's? (See also: https://docs.peppol.eu/poacc/billing/3.0/rules/UBL-CR-660/)
2. Same with `InvoiceLine`/`Price` `unitCodeListID` attribute: Is this absolutely necessary and applicable to all UBL's? https://docs.peppol.eu/poacc/billing/3.0/rules/UBL-CR-663/

Or does this have to do with the fact that the EN16931Test sets the `setCustomizationID` to `urn:cen.eu:en16931:2017`, and the `price`/`InvoiceLine` tags should *not* set the `unitCodeListID` and listID` attributes in that case? @p1eterdeboer do you have any input/pointers, as you wrote the EN16931 additions? 
Thanks!

JorisDebonnet commented:
Hi, thanks for reviewing. Yeah, I added a few attributes while I was trying to make our UBL compatible for a client, and I was quite confused between formats that require those attributes, while others warn against using them, ... because those were only warnings, not errors, I left them in. But it's possible they are not actually needed, so I removed them now. We'll see whether our client complains :)

brtdv commented:
OK thank you for the clarification. I'll merge these changes ASAP.

Yeah no worries, merging will work. Check out [contributing.md](https://github.com/num-num/ubl-invoice/blob/master/CONTRIBUTING.md) for more information.

p1eterdeboer commented:
For testing the EN16931 I've based this on the online check http://peppol.helger.com. So I'm assuming that the checks there are correct and have no other info on listID and unitCodeListID.

brtdv commented:
Hi @JorisDebonnet, been a while but I finally found some time to thoroughly check your PR. 

I split out the different listID and unitCodeListID attributes that can be used in UBL.be documents. Apparently when these are included in an EN16931 ubl document, the validation these documents fails. So I've made them optional. See also `SimpleUBL22InvoiceTest` for usage (`setListID` and `setUnitCodeListID`)
